src/jit/codegenarmarch.cpp

```diff
@@ -1617,11 +1766,29 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
         }
         else
         {
-            regNumber argReg = curArgTabEntry->regNum;
             genConsumeReg(argNode);
```

Comment:
It seems that you are still consuming only the first register here. Perhaps you need a `genConsumePutArgSplit` to consume all the register operands?